proto/main.proto

```diff
@@ -0,0 +1,113 @@
+edition = "2023";
+
+import "google/protobuf/empty.proto";
+
+service GoRemote {
+  // Start a new process, returning its ID.
+  rpc Start(StartRequest) returns (StartResponse) {}
+
+  // Stop a process by its ID.
+  rpc Stop(StopRequest) returns (google.protobuf.Empty) {}
+
+  // Check the status of a process by its ID.
+  rpc Status(StatusRequest) returns (StatusResponse) {}
+
+  // Stream the output of a process by its ID since the beginning of the
+  // process.
+  rpc Stream(StatusRequest) returns (stream Output) {}
+}
+
+// Ideally, I would use a type alias for all the places `uint32` is
+// used to refer to process IDs to improve readability/maintainability,
+// but unfortunately gRPC doesn't and won't support type aliases; see
+// <https://github.com/protocolbuffers/protobuf/issues/3521>. Also, it might
+// be nice to use a string to give processes human-readable names, but integers
+// are used instead because it will be simpler to implement since the server can
+// assign IDs so we don't have to handle reused names for example.
```

Comment:
On the topic of IDs, could you define how the server will generate them? Random, increasing, etc?